internal/fetcher/image_config/client.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2023 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package image_config
+
+import (
+	"context"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+)
+
+type key string
+
+const clientKey key = "ec.fetcher.config.client"
+
+type Client interface {
```

Comment:
We have so many clients, could we name this to be more specific, does it need to be public?
```suggestion
const imageFetcherKey key = "ec.fetcher.config.client"

type imageFetcher interface {
```

Or even:

```suggestion
const imageFetcherKey key = "ec.fetcher.config.client"

type imageFetcher func(name.Reference, ...remote.Option) (v1.Image, error)
```

Reply:
It doesn't have to be public actually. It is odd though because it is part of functions that are exported. :shrug: 

> We have so many clients

I don't see the problem with this. They are scoped to different packages. I find it beneficial to have a nice repeatable pattern.